### PR TITLE
fix: #49 Rebuilding test docker image on every run

### LINKED ISSUE #49: Pull the couchdb image directly , dont build the image to run the test
https://github.com/trustbloc/hub-store/blob/9a727f0015a491ab1553092fd775fedc7995d2bd/Makefile#L31-L36

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 9a727f0015a491ab1553092fd775fedc7995d2bd when #35 was merged. cc @talwinder50.

### PR DESCRIPTION
**Pull request by llorllale**
Fixes #49 

Test code now pulls docker image and configures it prior to running the tests.

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: internal/server/collection/couchdb/couchdbsetup_test.go [hunk 1]
```diff
@@ -7,39 +7,52 @@ SPDX-License-Identifier: Apache-2.0
...
+	ctx, _ := context.WithTimeout(context.Background(), timeout)
+	err = client.PullImage(
+		docker.PullImageOptions{
+			Context: ctx,
+			Repository: strings.Replace(image, ":", "@", 1),
+		},
+		docker.AuthConfiguration{},
+	)
+	if err != nil {
+		return errors.Wrapf(err, "cannot pull couchdb image %s", image)
```
Comment (rolsonquadras) [inline internal/server/collection/couchdb/couchdbsetup_test.go]: we can remove "couchdb" in the error message as 'image' var will log the image name.

File: internal/server/collection/couchdb/couchdb_test.go [hunk 2]
```diff
@@ -28,15 +29,27 @@ import (
...
+	if err := CreateIndices(url, dbname, 10 * time.Second); err != nil {
+		fmt.Printf("cannot create couchdb indices: err=%s", err)
+		cleanup()
+		os.Exit(1)
 	}
-	cfg := &Config{URL: url, DBName: dbname}
-	s = Store(cfg)
+	s = Store(&Config{URL: url, DBName: dbname})
+	code := m.Run()
+	cleanup()
```
[the remark this answers is not in the document]
Comment (llorllale) [reply internal/server/collection/couchdb/couchdb_test.go]: @rolsonquadras [deferred functions are not run when `os.Exit()` is called](https://golang.org/pkg/os/#Exit)

File: internal/server/collection/couchdb/couchdbsetup_test.go [hunk 3]
```diff
@@ -7,39 +7,52 @@ SPDX-License-Identifier: Apache-2.0
...
 
 	docker "github.com/fsouza/go-dockerclient"
 	"github.com/google/uuid"
 	"github.com/pkg/errors"
-	"github.com/thedevsaddam/gojsonq"
 )
 
-func couchdbImageName() string {
-	return os.Getenv("HUB_STORE_COUCHDB_IMAGE")
+func Pull(image string, timeout time.Duration) error {
```
[the remark this answers is not in the document]
Comment (llorllale) [reply internal/server/collection/couchdb/couchdbsetup_test.go]: @rolsonquadras this is a wrapper around `fsouza/go-dockerclient`, aka. the *real* generic util

File: internal/server/collection/couchdb/couchdbsetup_test.go (hunk 1 above)
Comment (llorllale) [reply internal/server/collection/couchdb/couchdbsetup_test.go]: @rolsonquadras fixed

File: internal/server/collection/couchdb/couchdbsetup_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rolsonquadras) [reply internal/server/collection/couchdb/couchdbsetup_test.go]: @llorllale i would say argue that the wrapper itself is generic and can be reused to pull any image.

**Pull request merged**